src/clients/components/ThemeSwitch.tsx

```diff
@@ -0,0 +1,32 @@
+import {
+  Switch,
+  Group,
+  useMantineColorScheme,
+  useMantineTheme,
+} from "@mantine/core";
+import { IconSun, IconMoonStars } from "@tabler/icons-react";
+
+const ThemeSwitch = () => {
```

Comment:
> I am afraid we have some duplicated codes because I also created a theme provider and a theme toggle button this time. But don't worry , I used same reference as you mentioned in description so most of them look similar. I can just combine them and delete duplicated codes after I merged yours!

Oh apologies for not having enough conversation about that.
I just included all code related to Header and Footer basically...

Reply:
No worries. It's all fine!
I should've noticed the toggle button belongs to Header as well.